rfds/116-Handling-Old-Comments/index.adoc

```diff
@@ -0,0 +1,276 @@
+ base_branch | text                     |           | not null | 
+ base_sha    | text                     |           | not null | 
+ branch      | text                     |           | not null | 
+ pull_number | bigint                   |           | not null | 
+ repository  | bigint                   |           | not null | 
+ sha         | text                     |           | not null | 
+ state       | text                     |           | not null | 
+ merged_sha  | text                     |           |          | 
+ merged_at   | timestamp with time zone |           |          | 
+ title       | text                     |           |          | 
+ username    | text                     |           |          | 
+----
+
+* `github_pull_requests_map`
+
+[source,sql]
+----
+terrateam=> \d github_pull_requests_map
+              Table "public.github_pull_requests_map"
+    Column     |  Type  | Collation | Nullable |      Default      
+---------------+--------+-----------+----------+-------------------
+ repository_id | bigint |           | not null | 
+ pull_number   | bigint |           | not null | 
+ core_id       | uuid   |           | not null | gen_random_uuid()
+----
+
+=== Terrateam Configuration
+
+. Given that our aim to make the most configurable IaC tool ever, we need to embed a new set of configurations and default values.
+. Focus on making it per-directory so people can have freedom to apply different commenting rules.
+
+== Proposals
+
+Our proposal is to allow users to select different strategies and give then options to apply them separetely, as this will cover a large amount of use cases.
+
+=== Strategies for Comment Handling
+
+Immutable (Current Behavior) :: Append a new comment for each operation
+* Pros: Simplest one to implement, audit-friendly.
+* Cons: Leads to PR clutter and makes status tracking harder.
+
+Minimize (or Mark Comments as Outdated) :: Use GitHub’s API to mark earlier comments as outdated.
+* Pros: Reduces visible clutter, users can still fix data if something bad happens.
+* Cons: Can be confusing if related issues are obscured.
+
+Delete :: Remove previous comments entirely once new ones are posted.
+* Pros: Cleanest PR view.
+* Cons: Loss of historical context, users can't manually fix stuff if something bad happens.
+
+Update (Re-use Comments) :: Link comments to a specific directory (dirspace) via metadata in the work_manifest, and update them in-place.
+* Pros: Clean UI, less noisy. Once inplemented, summaries are just a corollary of updates.
+* Cons: Requires properly tracking comment IDs to `work_manifests`. Hardest one to implement.
+
+==== Terrateam Configuration
+
+The following is a proposal of how that would look like as a top level key:
+
+[source,yaml]
+----
+# Default configuration that solves 80% of customer pain
+comments:
+  enabled: true
+  policies:
+    - tag_query: ""
+      # immutable, minimize, delete, update
+      strategy: minimize
```

Comment:
My two cents: make this `update_strategy` and change `immutable` to `append`.  I could see us having other commenting features that might have a strategy.

But I 100% agree with your default config.